### PR TITLE
Configure new dry-system component_dirs

### PR DESCRIPTION
**Pull request by timriley**
This PR updates the dry-system container setup to use the new `component_dirs` setting introduced in https://github.com/dry-rb/dry-system/pull/155 and https://github.com/dry-rb/dry-system/pull/157

### CODE REVIEW

File: lib/hanami/slice.rb
Comment (jodosha) [inline lib/hanami/slice.rb:106]: Just a minor thing, shouldn't we check `root.directory?` as we do [here](https://github.com/hanami/hanami/pull/1093/files#diff-821c48c88c392a3a4fb873393e0c3c8401ca598885b47e1cfe8c0d4c57932293R253)?

File: lib/hanami/slice.rb
Comment (jodosha) [inline lib/hanami/slice.rb:110]: Same here.

File: lib/hanami/slice.rb
Comment (jodosha) [inline lib/hanami/slice.rb:112]: This line assumes UNIX like systems, should we use `::FIle::SEPARATOR` instead? This to guarantee cross platform compatibility.

Regarding the dot, do we have a central place where this namespace separator is defined and it can be referenced from here? This is both for maintenance perspective and also code clarity.

**Review by jodosha — APPROVED** [review]

File: lib/hanami/slice.rb
Comment (timriley) [reply lib/hanami/slice.rb:112]: Thanks @jodosha — I'm updating this now.

As for the dot, I've changed it to use `config.namespace_separator` (so it'll use the default value provided by dry-system), and added this card to make it possible for this config (and others) to be propagated to slices if the user configures something particular on the application container: https://trello.com/c/nQBwnUUX

**Comment by timriley** [comment]
This is now ready to go (thanks for the feedback, @jodosha!), but I'll wait for the next dry-system release (which I plan to make in the next day or two) before I merge, so I can have it depend on a proper release rather than dry-system master.

**Pull request merged**
